tests/conftest.py

```diff
@@ -59,6 +62,38 @@ def logger():
     return pyscaffold.log.logger
 
 
+@pytest.fixture
+def isolated_logger(logger):
```

Comment:
For future contributors (and future me ;-) I think it makes sense to describe better what is done here conceptually. As far as I see it you solve the whole singleton logging issue by creating multiple ones with unique ids, right? Somewhere it should be described.